Others/TowerOfHanoi.java

```diff
@@ -5,16 +5,13 @@
 class TowerOfHanoi {
     public static void shift(int n, String startPole, String intermediatePole, String endPole) {
         // if n becomes zero the program returns thus ending the loop.
-        if (n == 0) {
```

Comment:
Why are you removing the base case?

Reply:
recursion function will be terminated when `if (n != 0 )` failed